[Messenger] Add `TransportNamesStamp` to change the transport while dispatching a message (#39306)
symfony/symfony: src/Symfony/Component/Messenger/CHANGELOG.md

 * Add `SerializedMessageStamp` to avoid serializing a message when a retry occurs.
 * Automatically resolve handled message type when method different from `__invoke` is used as handler.
 * Allow `#[AsMessageHandler]` attribute on methods.

>  * Added capability for modifying transport layer while dispatching message

Reviewer: ```suggestion
 * Add `OverrideSendersStamp` to change the transport while dispatching a message
```
